ui/app/AppLayouts/Profile/stores/ContactsStore.qml

```diff
@@ -15,10 +15,15 @@ QtObject {
 
     // contactsModel holds all available contacts
     property var contactsModel: contactsModule.contactsModel
-    property var myContactsModel: contactsModule.myMutualContactsModel
-    property var blockedContactsModel: contactsModule.blockedContactsModel
-    property var receivedContactRequestsModel: contactsModule.receivedContactRequestsModel
-    property var sentContactRequestsModel: contactsModule.sentContactRequestsModel
+
+    readonly property var contactsModelAdaptor: ContactsModelAdaptor {
+        allContacts: contactsModel
+    }
+
+    property var myContactsModel: contactsModelAdaptor.mutualContacts
+    property var blockedContactsModel: contactsModelAdaptor.blockedContacts
+    property var receivedContactRequestsModel: contactsModelAdaptor.pendingReceivedRequestContacts
+    property var sentContactRequestsModel: contactsModelAdaptor.pendingSentRequestContacts
```

Comment:
According to the guideline, the store should not keep anything extra, especially adaptors. I know, it may lead to more refactor work but probably it's inevitable anyways (bc of the problems noted in the main comment).

Even if tt's tempting to keep it here now, it has negative consequences, especially for testability.